src/sql/azdata.proposed.d.ts

```diff
@@ -823,11 +823,61 @@ declare module 'azdata' {
 		parentTypeName?: string;
 	}
 
+	/**
+	 * Represents a selected range in the result grid.
+	 */
+	export interface SelectionRange {
+		fromRow: number;
+		toRow: number;
+		fromCell: number;
+		toCell: number;
+	}
+
+	/**
+	 * Parameters for the copy results request.
+	 */
+	export interface CopyResultsRequestParams {
+		/**
+		 * ownerUri of the editor
+		 */
+		ownerUri: string;
+		/**
+		 * Index of the batch.
+		 */
+		batchIndex: number;
+		/**
+		 * Index of the result set.
+		 */
+		resultSetIndex: number;
+		/**
+		 * Whether to include the column headers.
+		 */
+		includeHeaders: boolean
+		/**
+		 * Whether to remove line breaks from the cell value.
+		 */
+		removeNewLines: boolean;
```

Comment:
Shouldn't this be something that the provider handles determining itself? I suppose maybe this is convenient for passing along the Copy Remove New Lines setting...is that the intention?

Reply:
this is a setting user can set in ADS, need to pass it to the backend.

Reply:
Yeah, but backend providers can get it themselves if needed too so I was wondering if it's actually necessary.

Or could we even consider having the logic to do this in core instead? That way the providers wouldn't even have to concern themselves with it.

Reply:
The copy to clipboard is handled by the provider, ADS core doesn't receive the content at all, so we have to pass the flag and let the provider handler it.

Reply:
Any reason why we'd want to do it that way? Is it to save having to transfer the copy content back to core from the extension?